woocommerce/woocommerce
# Add unique_id field to product

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Adds a new "unique id" field to the new product editor. The field accepts any type of input, at least for now.

Closes #41816
Closes #39577








### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. If you're running locally, please deactivate and activate the WooCommerce plugin, as this adds a new column in the wp_wc_product_meta_lookup table.
2. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
3. Go to Products > Add
4. Go to the 'Inventory' tab
5. Check the field labeled "GTIN, UPC, EAN OR ISBN"
6. You should be able to type any string into it.
7. It should save correctly.
8. Create another product
9. Fill in the name, and try to type the same value as you typed before for GTIN, UPC, EAN OR ISBN
10. Try to publish
11. An error should occur, as we don't allow duplicated values.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
